### PR TITLE
Remove optional 'text/css' attribute from HTML templates

### PR DESCRIPTION
**Pull request by MrBrain295**
This removes the `type='text/css'` from the style elements. According to [here](https://www.h3xed.com/web-development/is-type-text-css-needed-for-style-and-link-tags) it is unnecessary.

### CODE REVIEW

**Comment by bruvzg** [comment]
Is it causing issues? Even if it's optional (link to the W3C standard, instead of random blog would be appreciated), what's the benefit from removal?

**Comment by MrBrain295** [comment]
https://html.spec.whatwg.org/multipage/semantics.html#the-style-element

**Comment by MrBrain295** [comment]
Its not a problem I guess, just normally following the spec is best.

**Comment by SideFli** [comment]
This PR looks good!
 
He is right ngl.

**Review by CasualSideo — APPROVED** [review]
I like this! the code got improved.

**Review by Calinou — APPROVED** [review]
I removed it from `editor.html` recently, so it makes sense to remove it from the exported project template HTMLs as well.

**Comment by akien-mga** [comment]
Please squash the two commits into one, and edit the commit message to be more explicit (e.g. "Remove optional 'text/css' attribution in HTML templates"). See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) for instructions.

**Comment by MrBrain295** [comment]
I tried to find something about how to squash commits but I can’t find anything for GitHub and I don’t have a local fork.

**Comment by MrBrain295** [comment]
Sorry, I have been trying for 2 hours to squash the commits and it can't get them to squash.

**Comment by pycbouh** [comment]
What exactly goes wrong for you? You will need to squash your commits for the other PR as well if it’s approved https://github.com/godotengine/godot/pull/47083, so you better learn how to do it now.

**Comment by LikeLakers2** [comment]
@MrBrain295 Perhaps this may help: https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase

(P.S. @pycbouh You should only comment if you actually have something to add; also, saying "You better figure it out" helps nobody, and only makes you seem rude.)

**Comment by pycbouh** [comment]
@LikeLakers2 Instead of guessing, I've asked, verbatim, "What exactly goes wrong for you?", because otherwise we cannot help our potential contributor fix their issue. "You better learn how to do it now" is an encouragement to tackle the issue now and don't rely on maintainers merging the PR in its current state.

You, on the other hand, provided the same link as Akien already did, and then blamed me for being rude. This is unprofessional and has no place in our public work environment.

**Comment by LikeLakers2** [comment]
@pycbouh Ah, my apologies. I actually didn't notice Akien's link -- and even if I did, I probably would not have inspected it enough to notice that it links to the same section. But still, my apologies.

**Comment by akien-mga** [comment]
To clarify, it's not possible to squash or do any kind of advanced editing of Pull Requests using the online GitHub web editor.

That's why our documented [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) is based on command line Git with a local Git clone (it's also possible to use Git GUI with a local Git clone).

Using the GitHub web editor is fine for contributions to some projects/repository, but Godot is a huge codebase with thousands of contributors and we have stricter requirements to ensure that developers can work efficiently. That's why we require small changes which don't impact the dev workflow to be self-contained in a single commit that does the relevant changes, so that it doesn't take the focus away from other commits which have a significantly stronger impact on all users and contributors.

When contributing to an open source project, it's always good to review how existing contributors work, both from their documented workflow (PR workflow page, `CONTRIBUTING.md`) and the existing commit history to see how things typically look like: https://github.com/godotengine/godot/commits/
What kind of language/format is used in commit messages? How big are commits in terms of impact they have on the codebase compared to other projects? Etc.

Note: This is in no way a criticism, nor intended only for @MrBrain295, this is just additional context on why we have specific requirements and why some extra work is put on the contributors' side instead of the maintainers' side.

That being said, I'm aware that the path from the current state to a "correct" PR with changes squashed and including #47083 is a steep learning curve for a contributor not experienced with Git, so I'll amend the PR myself. For future PRs, I suggest reading our PR workflow documentation and trying that workflow out, instead of using the GitHub web editor.

**Comment by akien-mga** [comment]
Updated with a squashed commit including #47083 and a clearer commit message: acb98299d8bd255cb961b1c719e7072225a50960.

That being done, from what I've read in the spec and on the linked stackoverflow answer, I'm not sure this is actually a useful change.
Yes, the `type` attribute can be omitted because the browsers can guess it from the actual file's MIME type... but we can also save them the guesswork which doesn't seem to have any negative impacts?

I'll leave it up to @Faless to decide as he's the maintainer of the HTML5 platform.

**Comment by Faless** [comment]
I would rather not do this for the export template, where compatibility is more important even when servers report bogus mime types (which reminds me, I should also force the mime type when fetching the `wasm`).

**Comment by MrBrain295** [comment]
@Faless should I add `type=text/css` from `editor.html`?

**Comment by Faless** [comment]
> should I add `type=text/css` to `editor.html`?

no, the editor requires proper server headers configuration anyway (due to COOP/COEP requirements).

**Pull request closed without merging**
